### PR TITLE
Fixing escaping for multiple commas in csv

### PR DESCRIPTION
**Pull request by dsgouda**
Fixed a bug where the escaping logic would not take into consideration strings with multiple commas in them
@davmason @brianrob

### CODE REVIEW

File: src/xunit.performance.analysis/Program.cs [hunk 1]
```diff
@@ -287,14 +287,17 @@ private static void WriteTestResultsHtml(Dictionary<string, Dictionary<string, T
...
         {
             // Escape the csv string
-            if(str.Contains("\""))
+            if (str.Contains("\""))
             {
-                str = "\"" + str.Replace("\"", "\"\"") + "\"";
+                str = str.Replace("\"", "\"\"");
             }
-            if (str.Contains(","))
+            int numOfCommas = str.Split(',').Length - 1;
```
Comment (pharring) [inline src/xunit.performance.analysis/Program.cs]: There's got to be a better way. String.Split is a memory hog.
Is the idea to add a pair of enclosing quotes for each comma?

```C#
using System.Linq;
int numOfCommas = str.Count(ch => ch == ',');
string enclosingQuotes = new string('\"', numOfCommas);
return enclosingQuotes + str + enclosingQuotes;
```

Comment (dsgouda) [inline src/xunit.performance.analysis/Program.cs]: Nice! Will implement this. Yes, I am enclosing the string as many times as the number of commas.

**Comment by davmason** [comment]
Why do you need to add a quote for every comma? Shouldn't one set of quotes be sufficient?

Also, it would be nice if you could rebase the multiple commits in to one before checking in.

**Comment by dsgouda** [comment]
That is what it did earlier but that doesn't apparently work when we have strings with multiple commas.

**Comment by davmason** [comment]
LGTM

**Pull request merged**